File: getRandom.ts
function getRandom = (type_: string, length: number) => {
    let result = "";
    const type = type_.toLowerCase();
    length = length ? length : 32;
    let characters = "0123456789";
    if (type === 'number' || type === 'numbers') {
      characters = "0123456789";
      for (let i = 0; i < length; i++) {
        result += characters[Math.floor(Math.random() * characters.length)];
      }
      return result;
    } else if (type === 'alphabet' || type === 'alphabets') { 
      characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
      for (let i = 0; i < length; i++) {
        result += characters[Math.floor(Math.random() * characters.length)];
      }
      return result;
    } else if (type === 'alphanumeric' || type === 'all') {
      characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
      for (let i = 0; i < length; i++) {
        result += characters[Math.floor(Math.random() * characters.length)];
      }
      return result;
    } else {
      characters = type;
      for (let i = 0; i < length; i++) {
        result += characters[Math.floor(Math.random() * characters.length)];
      }
      return result;
    }
}

File: java/getRandom.java
package com.secure.chat.utils;

import java.security.SecureRandom;

public class GetRandom {
    private static final String NUMERIC = "0123456789";
    private static final String LOWERCASE = "abcdefghijklmnopqrstuvwxyz";
    private static final String UPPERCASE = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    private static final String ALPHANUMERIC = NUMERIC + LOWERCASE + UPPERCASE;

    private String getCharacters(String type) {
        return switch (type) {
            case "numeric" -> NUMERIC;
            case "lowercase" -> LOWERCASE;
            case "uppercase" -> UPPERCASE;
            case "all", "alphanumeric" -> ALPHANUMERIC;
            default -> throw new IllegalArgumentException("Invalid type: " + type);
        };

//        if (type.equals("numeric")) return NUMERIC;
//        else if (type.equals("lowercase")) return LOWERCASE;
//        else if (type.equals("uppercase")) return UPPERCASE;
//        else if (type.equals("all") || type.equals("alphanumeric")) return ALPHANUMERIC;
//        else throw new IllegalArgumentException("Invalid type: " + type);
    }

    public String getRandom(String type, int length) {
        StringBuilder result = new StringBuilder();
        String characters = getCharacters(type);
        SecureRandom random = new SecureRandom();

        for (int i = 0; i < length; i++) {
            int randomIndex = random.nextInt(characters.length());
            result.append(characters.charAt(randomIndex));
        }

        return result.toString();
    }
}
